### PR TITLE
initialize conditions at start and mirror keystoneservice condition

### PR DESCRIPTION
**Pull request by stuggi**
* Initializes the used conditions if the conditions are nil to their Status=Unknown.
* Add keystoneservice condition using condition.mirror(t type)
* Also adds a missing ExposeServiceReadyCondition MarkTrue

### CODE REVIEW

**Comment by gibizer** [comment]
/lgtm
I think after all these refactors the condition handling became pretty nice. Great job!

**Review by abays — APPROVED** [review]
/lgtm

**Comment by stuggi** [comment]
> /lgtm I think after all these refactors the condition handling became pretty nice. Great job!

Thanks, yes I also like the result. I think it makes the condition handling easy. Hope others see it the same when they start to use it.

File: controllers/placementapi_controller.go [hunk 1]
```diff
@@ -128,7 +128,10 @@ func (r *PlacementAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			condition.UnknownCondition(condition.ExposeServiceReadyCondition, condition.InitReason, condition.ExposeServiceReadyInitMessage),
 			condition.UnknownCondition(condition.InputReadyCondition, condition.InitReason, condition.InputReadyInitMessage),
 			condition.UnknownCondition(condition.ServiceConfigReadyCondition, condition.InitReason, condition.ServiceConfigReadyInitMessage),
-			condition.UnknownCondition(condition.DeploymentReadyCondition, condition.InitReason, condition.DeploymentReadyInitMessage))
+			condition.UnknownCondition(condition.DeploymentReadyCondition, condition.InitReason, condition.DeploymentReadyInitMessage),
+			// right now we have no dedicated KeystoneServiceReadyInitMessage
+			condition.UnknownCondition(condition.KeystoneServiceReadyCondition, condition.InitReason, ""),
+		)
```
Comment (bogdando) [inline controllers/placementapi_controller.go:134]: late to the party, but shouldn't the message be like "Deployment status unknown"?

**Pull request merged**